mgmt: mcumgr: Change transports from select to depends on

nordicjm opened the pull request:
This changes some of the MCUmgr transport CMake select statements
    to depends on, this is to align with other in-tree symbols that
    do not pull in whole subsystems and instead only allow selection
    if those subsystems are already enabled.

Fixes #53777

nordicjm committed:
mgmt: mcumgr: Change transport select to depends on

This changes some of the MCUmgr transport CMake select statements
to depends on, this is to align with other in-tree symbols that
do not pull in whole subsystems and instead only allow selection
if those subsystems are already enabled.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
samples: mgmt: mcumgr: Select required symbols

Select required dependency symbolds for the various transports.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

de-nordic reviewed (dismissed):
Looks OK.
Shouldn't this been given release notes entry? There is bug report and this requires some changes in user code (for example addition of CONFIG_BT=y if it has previously been missing because CONFIG_MCUMGR_TRANSPORT_BT would add that).

nordicjm committed:
doc: release: 3.3: Add note on MCUmgr transports

Adds details on additional Kconfigs that need to be selected for
MCUmgr transports.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>
